python/cudf/cudf/bindings/copying.pyx

```diff
@@ -39,26 +43,48 @@ def clone_columns_with_size(in_cols, row_size):
     return out_cols
 
 
-def apply_gather(in_cols, maps, out_cols=None):
-    """
-      Call cudf::gather.
+def _normalize_maps(maps, size):
+    maps = columnops.as_column(maps).astype("int32")
+    maps = maps.binary_operator("mod", maps.normalize_binop_value(size))
```

Comment:
Do we want the C++ mod or the Python mod here? If maps is negative this will keep it negative with a C++ modulo while a Python modulo would make it positive.

Reply:
`"mod"` is a Python modulo:

https://github.com/rapidsai/cudf/blob/branch-0.9/python/cudf/cudf/bindings/binops.pyx#L27

Reply:
TIL, thanks!